Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm

```diff
@@ -4025,18 +4037,63 @@ void determineChangedTracksFromNewTracksAndOldItems(MediaSelectionGroupAVFObjC*
 }
 
 #if HAVE(SPATIAL_TRACKING_LABEL)
+const String& MediaPlayerPrivateAVFoundationObjC::defaultSpatialTrackingLabel() const
+{
+    return m_defaultSpatialTrackingLabel;
+}
+
+void MediaPlayerPrivateAVFoundationObjC::setDefaultSpatialTrackingLabel(const String& defaultSpatialTrackingLabel)
+{
+    if (m_defaultSpatialTrackingLabel == defaultSpatialTrackingLabel)
+        return;
+    m_defaultSpatialTrackingLabel = WTFMove(defaultSpatialTrackingLabel);
+    updateSpatialTrackingLabel();
+}
+
 const String& MediaPlayerPrivateAVFoundationObjC::spatialTrackingLabel() const
 {
     return m_spatialTrackingLabel;
 }
 
-void MediaPlayerPrivateAVFoundationObjC::setSpatialTrackingLabel(String&& spatialTrackingLabel)
+void MediaPlayerPrivateAVFoundationObjC::setSpatialTrackingLabel(const String& spatialTrackingLabel)
 {
     if (m_spatialTrackingLabel == spatialTrackingLabel)
         return;
     m_spatialTrackingLabel = WTFMove(spatialTrackingLabel);
-    if (m_avPlayer)
+    updateSpatialTrackingLabel();
+}
+
+void MediaPlayerPrivateAVFoundationObjC::updateSpatialTrackingLabel()
+{
+    assertIsMainThread();
+
+    if (!m_avPlayer)
+        return;
+
+    if (!m_spatialTrackingLabel.isNull()) {
         [m_avPlayer _setSTSLabel:nsStringNilIfNull(m_spatialTrackingLabel)];
+        return;
+    }
+
+    AVAudioSession *session = [PAL::getAVAudioSessionClass() sharedInstance];
+    if (!m_visible) {
+        // If the page is not visible, use the AudioSession's STS label.
+        [m_avPlayer _setSTSLabel:session.spatialTrackingLabel];
+        return;
+    }
+
+    if (m_videoLayer) {
+        // Let AVPlayer manage setting the spatial tracking label in its AVPlayerLayer itself;
+        [m_avPlayer _setSTSLabel:nil];
+        return;
+    }
+
+    // If there is no AVPlayerLayer, use the default spatial tracking label if available, or
+    // the session's spatial tracking label if not.
+    if (!m_defaultSpatialTrackingLabel.isNull())
+        [m_avPlayer _setSTSLabel:nsStringNilIfNull(m_defaultSpatialTrackingLabel)];
```

Comment:
No need for `nsStringNilIfNull` if we're already checking for null on the line above.